doc/source/reference/c-api/float16.rst

```diff
@@ -0,0 +1,248 @@
+.. _float16:
+
+Half-precision API
+==================
+
+The header file ``<numpy/float16.h>`` provides helper routines for working
+with IEEE 754-2008 16-bit floating-point values
```

Comment:
Not sure where, but the `float16.h` header is special in the sense that none of the functions here require "importing" NumPy (since they don't call any functions).
I think it would be nice to mention that somewhere.

I guess we need `ndarraytypes.h` to have `npy_float16` and `npy_half` defined? Would be cool if we didn't need the full thing, or need it at all (it's rather large).
But I am not really worried about that.